File: assignment_01/original/two_sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        given arr nums, int target
        go through nums, save diff to map of idxes
        return idxes if it = target - num
        """
        idxes = {}
        for i in range(len(nums)):
            num = nums[i]
            diff = target - num
            if diff in idxes:
                return [i, idxes[diff]]
            idxes[num] = i

File: assignment_01/src/two_sum.cpp
#include "two_sum.hpp"
#include <unordered_map>

std::vector<int> twoSum(const std::vector<int>& nums, int target) {
    std::unordered_map<int, int> idxes;
    int n = nums.size();
    for (int i = 0; i < n; ++i) {
        int num = nums[i];
        int diff = target - num;
        if (idxes.find(diff) != idxes.end()) {
            return {i, idxes[diff]};
        }
        idxes[num] = i;
    }
    return {};
}
